### ISSUE TITLE
False positive with NewType wrapped in Annotated

### ISSUE THREAD

**Issue by minmax** [microsoft/pyright#8755]
Because of https://github.com/microsoft/pyright/issues/7370 I think this form of using `Annotated` should also be banned.

```python
from typing import Annotated, NewType, TypeVar

T = TypeVar("T")

Id = NewType("Id", int)

PkId = Annotated[T, ">=1"]

MyPkId = PkId[Id]

print(MyPkId)
# > typing.Annotated[__main__.Id, '>=1']

print(MyPkId(1)) # "Annotated cannot be instantiated" not raised

```

pyright 1.1.375

**Comment by erictraut** [comment]
I presume you meant "false negative" in the title. 

I don't think this is the same case as #7370. In this case, `MyPkId` is a specialized version of the type alias, so its type is divorced from the original `Annotated`, at least from a static type perspective.

I think this is an edge case and is not likely to result in any bugs, so I'm going to reject this for now. I will revisit this decision if and when draft [PEP 747](https://discuss.python.org/t/pep-747-typeexpr-type-hint-for-a-type-expression/55984/43) clarifies the intended behavior in this case.

**Comment by minmax** [comment]
I would like this loophole to remain myself.
But I also worry that in this case, additional code is executed at runtime:

```python
class _BaseGenericAlias(_Final, _root=True):
    def __call__(self, *args, **kwargs):
        ...
        result = self.__origin__(*args, **kwargs)
        try:
            result.__orig_class__ = self
        except AttributeError:
            pass
        return result
```

`__orig_class__` attribute setted for `1` on call `MyPkId(1)`.
